plugin/evm/block_builder.go

```diff
@@ -160,9 +142,31 @@ func (b *blockBuilder) awaitSubmittedTxs() {
 				log.Trace("New atomic Tx detected, trying to generate a block")
 				b.signalTxsReady()
 			case <-b.shutdownChan:
+				b.pendingSignal.Broadcast()
 				b.buildBlockTimer.Stop()
 				return
 			}
 		}
 	})
 }
+
+func (b *blockBuilder) waitForTxEnqueue(ctx context.Context) commonEng.Message {
```

Comment:
Since this function only returns either `0` or `commonEng.PendingTxs`, I feel like this should just return a bool.